### DISCUSSION TITLE
Does session persist after closing the app?

### DISCUSSION THREAD

**Discussion by albyrock87** [pendo-io/pendo-mobile-sdk#92] [Q&A] [answered]
Regarding `void StartSession(string visitorId, string accountId, Dictionary<string, object> visitorData, Dictionary<string, object> accountData)`, documentation [here](https://github.com/pendo-io/pendo-mobile-sdk/blob/master/api-documentation/xamarin-maui-apis.md#startsession) says:

> If a session **is already in progress**, the current session will terminate and a new session will begin

Does the session persist after closing and reopening the app?

Example:
1. User opens the app => invoke `Setup`
2. User tries to log in
3. Login successful => invoke `StartSession` and store the application authentication token
4. User closes the app via system app switcher
5. User opens the app => invoke `Setup`
6. User is already logged in because we still have a valid authentication token
7. => do we need to invoke `StartSession` here?

**Comment by noambartouv** [comment, marked as answer]
Hey @albyrock87. Yes you need to call `StartSession` in both scenarios (manual login and automatic logic). The session does not pressit when closing the app. Let us know if you have further qustions.

Note: You also have the option to monitor the session before the manual login process by starting a session with an anonmyous visitor (by passing in a null value as the visitorId) when the app launches. Then later calling `StartSesion` once again, when the visitor has identified themselves during log in. Anonymous visitors will be considered as separate additional MAU than the identified visitors. In the very near future we are rolling out a new full customer journey feature that allows examining the anonmymous and identified data generated by the same end user in a combined manner.

I might make an adjust to the documentaion regarding the line you quoted: `If a session is already in progress, the current session will terminate and a new session will begin`. This is refrencing the scenario where `StartSession` is called more than once since the launch of the application (such as anonymous to known or a change of visitor).

**Reply by albyrock87** [reply]
This clarifies everything. Thank you very much!
